web/stream/stream_test.go

```diff
@@ -0,0 +1,108 @@
+package stream_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/quickfeed/quickfeed/web/stream"
+)
+
+type Data struct {
+	Msg string
+}
+
+var messages = []Data{
+	{Msg: "Hello"},
+	{Msg: "World"},
+	{Msg: "Foo"},
+	{Msg: "Bar"},
+	{Msg: "Gandalf"},
+	{Msg: "Frodo"},
+	{Msg: "Bilbo"},
+	{Msg: "Radagast"},
+	{Msg: "Sauron"},
+	{Msg: "Gollum"},
+}
+
+func TestStream(t *testing.T) {
+	service := stream.NewService[Data]()
+
+	counter := uint32(0)
+
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(1000*time.Second))
+	defer cancel()
+	streams := make([]*mockStream[Data], 0)
+
+	wg := sync.WaitGroup{}
+	for i := 1; i < 10; i++ {
+		st := service.AddStream(uint64(1), newMockStream[Data](ctx, uint64(1), &counter))
+		streams = append(streams, st.(*mockStream[Data]))
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := st.Run()
+			t.Log(err)
```

Comment:
Once we feel comfortable that the test is working as expected, maybe we should remove the printing.